<?xml version="1.0"?>
<launch>
    <arg name="roomba"      default="roomba1" />
    <arg name="config_path" default="$(find place_recognition)/config" />
    <arg name="date"        default="20220630" />

    <!--map_server -->
    <arg name="is_map" default="true" />
    <group if="$(arg is_map)" >
        <node pkg="map_server" type="map_server" name="map" args="$(find multi_localizer)/config/map/static_map.yaml" />
    </group>

    <!-- equirectangular_convertor, rosbag -->
    <include file="$(find place_recognition)/launch/utils/equirectangular_convertor.launch" >
        <!-- default -->
        <arg name="is_bag"               value="true" />
        <arg name="theta_s_mode"         value="0" />
        <arg name="is_theta_s_visualize" value="false" />
        <arg name="is_rviz"              value="false" />
        <arg name="wait_time"            value="15.0" />

        <!-- param -->
        <arg name="roomba"               value="$(arg roomba)" />
        <arg name="date"                 value="$(arg date)" />
    </include>

    <!-- place_recognition -->
    <arg name="publish_img"  default="true" />
    <arg name="publish_pose" default="true" />
    <node pkg="place_recognition" type="place_recognition" name="place_recognition" output="screen" >
        <!-- remap -->
        <!-- <remap from="img_in" to="roomba5/equirectangular/image_raw" /> -->
        <remap from="img_in"      to="$(arg roomba)/camera/color/image_rect_color" />
        <remap from="img_out"     to="$(arg roomba)/output_image" />
        <remap from="pose_out"    to="$(arg roomba)/output_pose" />
        <remap from="pr_pose_out" to="$(arg roomba)/pr_pose" />

        <!-- params -->
        <param name="REFERENCE_FILE_NAME" value="$(find place_recognition)/reference_data/" />
        <param name="IMAGE_MODE"          value="rgb" />
        <param name="DETECTOR_MODE"       value="orb" />
        <param name="PUBLISH_IMG"         value="$(arg publish_img)" />
        <param name="PUBLISH_POSE"        value="$(arg publish_pose)" />
    </node>

    <!-- rviz -->
    <arg name="is_rviz" default="true" />
    <group if="$(arg is_rviz)" >
        <arg name="rviz_wait_time" default="16.0" />
        <node pkg="rviz" type="rviz" name="$(arg roomba)_rviz" args="-d $(arg config_path)/rviz/place_recognition/$(arg roomba).rviz" launch-prefix="bash -c 'sleep $(arg rviz_wait_time); $0 $@'"/>
    </group>

</launch>
